lib/finch.ex

```diff
@@ -351,12 +350,84 @@ defmodule Finch do
 
   See `request/3` for more detailed information.
   """
-  @spec request!(Request.t(), name(), keyword()) ::
+  @spec request!(Request.t(), name(), request_opts()) ::
           Response.t()
   def request!(%Request{} = req, name, opts \\ []) do
     case request(req, name, opts) do
       {:ok, resp} -> resp
       {:error, exception} -> raise exception
     end
   end
+
+  @doc """
+  Sends an HTTP request asynchronously, returning a request reference.
+
+  ## Responses
+
+  Response information is sent to the calling process as it is received
+  in `{ref, response}` tuples. Responses include:
+
+    * `{:status, status}` - HTTP response status
+    * `{:headers, headers}` - HTTP response headers
+    * `{:data, data}` - section of the HTTP response body
+    * `{:error, exception}` - an error occurred during the request
+    * `:done` - request has completed successfully
```

Comment:
Maybe we should have trailing headers as part of `:done`?

```suggestion
    * `{:done, trailing_headers}` - request has completed successfully
```

Thoughts?

Reply:
Unless this version is much harder to implement.

Reply:
I don’t think it would be much harder. Right now it is 1-1 with the responses that Mint.stream returns, which doesn’t differentiate in the response shape between normal and trailing headers. 

So the thing to consider is that we’ll receive both trailing headers and a `:done` from Mint. Do we want to cache trailing headers and emit them if present when we receive `:done`, or would we want to immediately omit `:done` with those headers immediately and ignore the done response when we get it?

In my opinion, it’s not worth treating them specially — we just document (like Mint does) that an additional :headers received at the end of the response just prior to :done are trailing headers.

Reply:
Agreed then. Let’s leave it as is!